llvm/include/llvm/IR/IRBuilder.h

```diff
@@ -1902,16 +1902,18 @@ class IRBuilderBase {
   }
 
   Value *CreateConstGEP2_32(Type *Ty, Value *Ptr, unsigned Idx0, unsigned Idx1,
-                            const Twine &Name = "") {
+                            const Twine &Name = "",
+                            GEPNoWrapFlags NWFlags = GEPNoWrapFlags::none()) {
     Value *Idxs[] = {
       ConstantInt::get(Type::getInt32Ty(Context), Idx0),
       ConstantInt::get(Type::getInt32Ty(Context), Idx1)
     };
 
-    if (auto *V = Folder.FoldGEP(Ty, Ptr, Idxs, GEPNoWrapFlags::none()))
+    if (auto *V =
+            Folder.FoldGEP(Ty, Ptr, Idxs, /*IsInBounds=*/NWFlags.isInBounds()))
```

Comment:
Shouldn't this be?
```suggestion
            Folder.FoldGEP(Ty, Ptr, Idxs, NWFlags))
```

Reply:
Ah yes good point - thanks for spotting!

I've updated with this change, as well as a couple of arch specific test updates that were missed before due to my build config.

Reply:
It looks like this one still passes `NWFlags.isInBounds()` instead of the full `NWFlags`?

Reply:
Yes, sorry about that - updated to pass the full `NWFlags` correctly (+ associated test updates).